Dev

Eluminent opened the pull request.

BrendanHu committed:
Create .gitattributes for lfs

BrendanHu committed:
Commit Unity starter

Character w/ movement

LillianHuang-dev committed:
add player movement

BrendanHu committed:
Add barebones main menu

- New scene w/ button to go to game scene
- TODO: Determine art style and make it look better, will also have to load saved games if we have enough time.

Yangjas4 committed:
placeholder sprites

Yangjas4 committed:
upscaled main character sprites

LillianHuang-dev committed:
add player sprite

Yangjas4 committed:
Update WBC-movement-horizontal-4xupscale.png

Yangjas4 committed:
Merge branch 'dev' of https://github.com/TommyHuang-dev/pirate-game-jam-16 into dev

LillianHuang-dev committed:
add dashing by holding space

LillianHuang-dev committed:
Merge branch 'dev' of github.com:TommyHuang-dev/pirate-game-jam-16 into dev

LillianHuang-dev commented on .plastic/plastic.selector:
```diff
@@ -0,0 +1,4 @@
+repository "pirate-game-jam-16@nadnerbb@cloud"
```
I think we can remove all these plastic files

BrendanHu committed:
Remove plastic

VCS greedy :(

TommyHuang-dev commented:
LGTM

BrendanHu committed:
Navigation and etc

- Navigation from one scene to another when touching a capillary
  - Todo: Restrict scene switching until objectives are met
- Crossfade animation for scene entry
  - Not final, but good groundwork for future transition work if we choose to do it
- Player entrance animation
  - Gets shot down from above and spins a few times

BrendanHu committed:
Small nav/animation update

- Moved player animations to correct file location
- Placeholder room now matches the NewRunRoom, with a white circle to show that scene changing works

BrendanHu committed:
Audio Manager and Nav Refactoring

- Singleton audio manager to play music and SFX
    - Fades tracks in and out (added main theme and WIP battle theme for testing)
- LevelLoader.cs should eventually handle all logic related to room movement (pass data from capillary to LoadNextLevel?)

BrendanHu committed:
Improved navigation

- Capillaries will now generate the types of rooms that they lead to
  - This includes basic enemies (id 2), elite enemies (id 3), events (id 4), and bosses (id 5)
  - Bosses currently occur every 4 rooms
  - Capillaries cannot give duplicate options (no 2x basic enemies or events, for example)

- Converted player and our basic room format into prefabs
  - These are now reusable across scenes. Changing the prefab changes it for every scene.

- Made placeholders for all room types

- A little more audio testing with new room structure, WIP

LillianHuang-dev committed:
create prefab folder

LillianHuang-dev committed:
add character cs back

LillianHuang-dev committed:
change movement settings

BrendanHu committed:
Nerf WBC

- Acceleration 22 -> 2.5
- maxMoveSpeed 4 -> 5

LillianHuang-dev committed:
fix movement variables in new run screen and nerf acceleration 2.5 -> 2

BrendanHu committed:
LevelLoader refactor

Don't have to drag level loaders to character anymore

BrendanHu committed:
Merge branch 'dev' of https://github.com/TommyHuang-dev/pirate-game-jam-16 into dev
